cita-executor/core/src/cita_executive.rs

```diff
@@ -256,50 +259,165 @@ impl<B: DB + 'static> CitaExecutive<B> {
         let amend_type = value.low_u32();
-        let result = TxExecResult { gas_left: init_gas };
         match amend_type {
             AMEND_ABI => {
                 if self.transact_set_abi(&(data.to_owned().unwrap())) {
-                    Ok(result)
+                    Ok(())
                 } else {
                     Err(ExecutionError::Internal("Account doesn't exist".to_owned()))
                 }
             }
             AMEND_CODE => {
                 if self.transact_set_code(&(data.to_owned().unwrap())) {
-                    Ok(result)
+                    Ok(())
                 } else {
                     Err(ExecutionError::Internal("Account doesn't exist".to_owned()))
                 }
             }
             AMEND_KV_H256 => {
                 if self.transact_set_kv_h256(&(data.to_owned().unwrap())) {
-                    Ok(result)
+                    Ok(())
                 } else {
                     Err(ExecutionError::Internal("Account doesn't exist".to_owned()))
                 }
             }
 
             AMEND_ACCOUNT_BALANCE => {
                 if self.transact_set_balance(&(data.to_owned().unwrap())) {
-                    Ok(result)
+                    Ok(())
                 } else {
                     Err(ExecutionError::Internal(
                         "Account doesn't exist or incomplete trie error".to_owned(),
                     ))
                 }
             }
 
-            _ => Ok(result),
+            _ => Ok(()),
         }
     }
+
+    fn create(
+        &mut self,
+        block_provider: Arc<BlockDataProvider>,
+        state_provider: Arc<RefCell<State<B>>>,
+        params: &VmExecParams,
+    ) -> Result<Receipt, ExecutionError> {
+        let evm_transaction = build_evm_transaction(params);
+        let evm_config = build_evm_config(self.env_info.gas_limit.clone().as_u64());
+        let evm_context = build_evm_context(&self.env_info);
+        let receipt = match cita_vm::exec(
+            block_provider,
+            state_provider,
+            evm_context,
+            evm_config,
+            evm_transaction,
+        ) {
+            Ok(evm_result) => build_receipt_with_ok(params.gas, evm_result),
+            Err(e) => build_receipt_with_err(e),
+        };
+        Ok(receipt)
+    }
+}
+
+fn build_evm_transaction(params: &VmExecParams) -> EVMTransaction {
+    EVMTransaction {
+        from: params.sender,
+        value: params.value,
+        gas_limit: params.gas.as_u64(),
+        gas_price: params.gas_price,
+        input: params.data.clone(),
+        to: params.to_address,
+        nonce: params.nonce,
+    }
+}
+
+fn build_evm_context(env_info: &EnvInfo) -> EVMContext {
+    EVMContext {
+        gas_limit: env_info.gas_limit.as_u64(),
+        coinbase: env_info.author,
+        number: U256::from(env_info.number),
+        timestamp: env_info.timestamp,
+        difficulty: env_info.difficulty,
+    }
+}
+
```

Comment:
`build_emv_info` may be better.

Reply:
This is a function to build EVMContext, and `EVMContext` is defined in `cita-vm`. I think `Context` is okay for its meaning.